server/db/usersDao.js

```diff
@@ -0,0 +1,87 @@
+const Pool = require('pg').Pool
+const pool = new Pool({
+    user: 'admin',
+    host: 'localhost',
+    database: 'spp',
+    password: 'password',
+    port: 5432,
+})
+
+// pool.query("DROP TABLE users");
+//TODO
+pool.query('CREATE TABLE IF NOT EXISTS users (\n' +
+    '\tuser_id serial PRIMARY KEY,\n' +
+    '\tusername VARCHAR ( 50 ) UNIQUE NOT NULL,\n' +
+    '\tpassword VARCHAR ( 50 ) NOT NULL,\n' +
+    '\temail VARCHAR ( 255 ) UNIQUE NOT NULL,\n' +
+    '\tcreated_on VARCHAR ( 255 ),\n' +
+    '        last_login TIMESTAMP \n' +
+    ');');
+//TODO
+//pool.query("INSERT INTO users (username, password, email, created_on) VALUES ('username', 'password', 'email', 'created_on')")
+
+const getUsers = (request, response) => {
+    pool.query('SELECT * FROM users ORDER BY user_id ASC', (error, results) => {
+        if (error) {
+            throw error
+        }
+        response.status(200).json(results.rows)
+    })
+}
+
+const getUserById = (request, response) => {
+    const id = parseInt(request.params.id)
+
+    pool.query('SELECT * FROM users WHERE id = $1', [id], (error, results) => {
+        if (error) {
+            throw error
+        }
+        response.status(200).json(results.rows)
+    })
+}
+
+const createUser = (request, response) => {
+    const {name, email} = request.body
+
+    pool.query('INSERT INTO users (name, email) VALUES ($1, $2)', [name, email], (error, results) => {
+        if (error) {
+            throw error
+        }
+        response.status(201).send(`User added with ID: ${result.insertId}`)
+    })
+}
+
+const updateUser = (request, response) => {
+    const id = parseInt(request.params.id)
+    const {name, email} = request.body
+
+    pool.query(
+        'UPDATE users SET name = $1, email = $2 WHERE id = $3',
+        [name, email, id],
+        (error, results) => {
+            if (error) {
+                throw error
+            }
+            response.status(200).send(`User modified with ID: ${id}`)
+        }
+    )
+}
+
+const deleteUser = (request, response) => {
+    const id = parseInt(request.params.id)
+
+    pool.query('DELETE FROM users WHERE id = $1', [id], (error, results) => {
+        if (error) {
+            throw error
+        }
+        response.status(200).send(`User deleted with ID: ${id}`)
+    })
+}
+
+module.exports = {
```

Comment:
Have to export functions to simplify access. 
Basically we locate lambdas in this file and pass them as routes handlers.